src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -4083,11 +4103,25 @@ void MarkPropertiesOnTypeHierarchy (ref ReflectionPatternContext reflectionConte
 						if (filter != null && !filter (property))
 							continue;
 
-						if (staticOnly) {
+						if ((bindingFlags & (BindingFlags.Instance | BindingFlags.Static)) == BindingFlags.Static) { 
 							if ((property.GetMethod != null) && !property.GetMethod.IsStatic) continue;
 							if ((property.SetMethod != null) && !property.SetMethod.IsStatic) continue;
 						}
 
+						if ((bindingFlags & (BindingFlags.Instance | BindingFlags.Static)) == BindingFlags.Instance) {
+							if ((property.GetMethod != null) && property.GetMethod.IsStatic) continue;
+							if ((property.SetMethod != null) && property.SetMethod.IsStatic) continue;
+						}
+						if ((bindingFlags & (BindingFlags.Public | BindingFlags.NonPublic)) == BindingFlags.Public) {
+							if ((property.GetMethod != null) && !property.GetMethod.IsPublic) continue;
```

Comment:
This is incorrect - property is considered public even if only one of set/get is public. Similarly for events.